iothub/device/samples/getting started/RoutingTutorial/SimulatedDevice/Program.cs

```diff
@@ -0,0 +1,192 @@
+                //   uncomment the call here and the return command, then run this application.
+                ReadOneRowFromFile();
+            }
+            else
+            {
+                // Send messages to the simulated device. Each message will contain a randomly generated 
+                //   Temperature and Humidity.
+                // The "level" of each message is set randomly to "storage", "critical", or "normal".
+                // The messages are routed to different endpoints depending on the level, temperature, and humidity.
+                //  This is set in the tutorial that goes with this sample: 
+                //  http://docs.microsoft.com/azure/iot-hub/tutorial-routing
+
+                Console.WriteLine("Routing Tutorial: Simulated device\n");
+                string myDeviceId = parameters.DeviceId;
+                string deviceKey = parameters.DeviceKey;
+                s_deviceClient = DeviceClient.Create(
+                    parameters.IotHubUri,
+                    new DeviceAuthenticationWithRegistrySymmetricKey(myDeviceId, deviceKey),
+                    TransportType.Mqtt);
+
+                using var cts = new CancellationTokenSource();
+                Task messages = SendDeviceToCloudMessagesAsync(myDeviceId, cts.Token);
+                Console.WriteLine("Press the Enter key to stop.");
+                Console.ReadLine();
+                await s_deviceClient.CloseAsync(cts.Token);
+                cts.Cancel();
+                await messages;
+
+            }
+        }
+
+        /// <summary> 
+        /// Send message to the Iot hub. This generates the object to be sent to the hub in the message.
+        /// </summary>
+        private static async Task SendDeviceToCloudMessagesAsync(string myDeviceId, CancellationToken token)
+        {
+            double minTemperature = 20;
+            double minHumidity = 60;
+            var rand = new Random();
+
+            while (!token.IsCancellationRequested)
+            {
+                double currentTemperature = minTemperature + rand.NextDouble() * 15;
+                double currentHumidity = minHumidity + rand.NextDouble() * 20;
+
+                string infoString;
+                string levelValue;
+
+                if (rand.NextDouble() > 0.7)
+                {
+                    if (rand.NextDouble() > 0.5)
+                    {
+                        levelValue = "critical";
+                        infoString = "This is a critical message.";
+                    }
+                    else
+                    {
+                        levelValue = "storage";
+                        infoString = "This is a storage message.";
+                    }
+                }
+                else
+                {
+                    levelValue = "normal";
+                    infoString = "This is a normal message.";
+                }
+
+                var telemetryDataPoint = new
```

Comment:
what is the data type of `telemetryDataPoint`?

Reply:
Anonymous type, seems to be fine in this instance